src/data.rs

```diff
@@ -211,6 +257,21 @@ impl DataObj<'_> {
             None
         }
     }
+
+    /// Sets a default value for the key.
+    ///
+    /// Notes
+    /// -----
+    /// Setting a default value for a [`DataArray`] is current a no-op because
+    /// of a API problem of OBS.
+    pub fn set_default<N: Into<ObsString>, T: FromDataItem, V: Into<T>>(
+        &mut self,
+        name: N,
+        value: V,
+    ) {
+        unsafe { T::set_default_unchecked(self.as_ptr_mut(), name.into(), value.into()) }
+    }
```

Comment:
This should get around that issue with `set_default::<_, i64, _>`

```suggestion
    pub fn set_default<T: FromDataItem>(
        &mut self,
        name: impl Into<ObsString>,
        value: impl Into<T>,
    ) {
        unsafe { T::set_default_unchecked(self.as_ptr_mut(), name.into(), value.into()) }
    }
```